test/e2e/setup/setup.go

```diff
@@ -77,22 +82,29 @@ func CreateTLSResources(namespace string, ctx *e2eutil.Context) error { //nolint
 		return err
 	}
 
-	// Create server key and certificate secret
-	cert, err := ioutil.ReadFile(path.Join(testDataDir, "server.crt"))
-	if err != nil {
-		return err
-	}
-	key, err := ioutil.ReadFile(path.Join(testDataDir, "server.key"))
-	if err != nil {
-		return err
-	}
-
-	certKeySecret := secret.Builder().
+	certKeySecretBuilder := secret.Builder().
 		SetName(tlsConfig.CertificateKeySecret.Name).
-		SetNamespace(namespace).
-		SetField("tls.crt", string(cert)).
-		SetField("tls.key", string(key)).
-		Build()
+		SetNamespace(namespace)
+
+	if secretType == CertKeyPair {
+		// Create server key and certificate secret
+		cert, err := ioutil.ReadFile(path.Join(testDataDir, "server.crt"))
+		if err != nil {
+			return err
+		}
+		key, err := ioutil.ReadFile(path.Join(testDataDir, "server.key"))
+		if err != nil {
+			return err
+		}
+		certKeySecretBuilder.SetField("tls.crt", string(cert)).SetField("tls.key", string(key))
+	} else if secretType == Pem {
```

Comment:
do we need an `else if` instead of just `if` , since the secretType can be only one ?